jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -50,6 +52,9 @@ class S3TreeTable(
     private val project: Project
 ) : TreeTable(treeTableModel), CoroutineScope by ApplicationThreadPoolScope("S3TreeTable") {
     private val edt = getCoroutineUiContext()
+    val bucketVersioningStatus: BucketVersioningStatus = runBlocking {
+        bucket.getBucketVersioning().status()
```

Comment:
can we grab this asynchronously instead? what happens if the user doesn't have IAM permissions to get the bucket versioning status?

Reply:
I removed this altogether since there is no clear solution for user that has IAM permissions to get object but just lacks permisions to check if bucket is versioned.
Call to get versions of object in bucket that is not versioned is still valid call that returns just single value.